core/app/models/spree/unit_cancel.rb

```diff
@@ -6,18 +6,18 @@ class Spree::UnitCancel < Spree::Base
   DEFAULT_REASON = 'Cancel'
 
   belongs_to :inventory_unit
-  has_one :adjustment, as: :source, dependent: :destroy
+  has_many :adjustments, as: :source, dependent: :destroy
 
   validates :inventory_unit, presence: true
 
   # Creates necessary cancel adjustments for the line item.
   def adjust!
-    raise "Adjustment is already created" if adjustment
+    raise "Adjustment is already created" if adjustments.exists?
 
     amount = compute_amount(inventory_unit.line_item)
 
-    create_adjustment!(
-      adjustable: inventory_unit.line_item,
+    inventory_unit.line_item.adjustments.create!(
```

Comment:
Alternatively, I think this could be.

```
self.adjustment = inventory_unit.line_item.adjustments.create!(
```

This should keep both associations fresh without performing an extra query (no query run since source will already be set correctly). This avoids the need to change the `has_one` to a `has_many`